cli/cppcheckexecutor.cpp

```diff
@@ -94,31 +102,52 @@ namespace {
                 if (ruleIds.insert(finding.id).second) {
                     picojson::object rule;
                     rule["id"] = picojson::value(finding.id);
+                    // rule.name
+                    rule["name"] = picojson::value(getRuleShortDescription(finding));
                     // rule.shortDescription.text
                     picojson::object shortDescription;
-                    shortDescription["text"] = picojson::value(finding.shortMessage());
+                    shortDescription["text"] = picojson::value(getRuleShortDescription(finding));
                     rule["shortDescription"] = picojson::value(shortDescription);
                     // rule.fullDescription.text
                     picojson::object fullDescription;
-                    fullDescription["text"] = picojson::value(finding.verboseMessage());
+                    fullDescription["text"] = picojson::value(getRuleFullDescription(finding));
                     rule["fullDescription"] = picojson::value(fullDescription);
                     // rule.help.text
                     picojson::object help;
-                    help["text"] = picojson::value(finding.verboseMessage()); // FIXME provide proper help text
+                    help["text"] = picojson::value(getRuleFullDescription(finding));
                     rule["help"] = picojson::value(help);
                     // rule.properties.precision, rule.properties.problem.severity
                     picojson::object properties;
                     properties["precision"] = picojson::value(sarifPrecision(finding));
-                    double securitySeverity = 0;
-                    if (finding.severity == Severity::error && !ErrorLogger::isCriticalErrorId(finding.id))
-                        securitySeverity = 9.9; // We see undefined behavior
-                    //else if (finding.severity == Severity::warning)
-                    //    securitySeverity = 5.1; // We see potential undefined behavior
-                    if (securitySeverity > 0.5) {
-                        properties["security-severity"] = picojson::value(securitySeverity);
-                        const picojson::array tags{picojson::value("security")};
-                        properties["tags"] = picojson::value(tags);
+                    // Only set security-severity for findings that are actually security-related
+                    if (isSecurityRelatedFinding(finding.id)) {
+                        double securitySeverity = 0;
+                        if (ErrorLogger::isCriticalErrorId(finding.id)) {
+                            securitySeverity = 9.9; // critical = 9.0+
```

Comment:
I am not sure.. a "critical id" means that cppcheck analysis is aborted so probably the cppcheck configuration is not ideal. 

It does not indicate at all if there is a security issue or not.

Is there some documentation that says how the security severity should be assigned in this case?